haystack/nodes/prompt/invocation_layer/anthropic_claude.py

```diff
@@ -0,0 +1,229 @@
+        kwargs includes: stop_sequences, temperature, top_p, top_k, and stream. For more details about these kwargs,
+        see Anthropic's [documentation](https://console.anthropic.com/docs/api/reference).
+        """
+        super().__init__(model_name_or_path)
+        if not isinstance(api_key, str) or len(api_key) == 0:
+            raise AnthropicError(
+                f"api_key {api_key} must be a valid Anthropic key. Visit https://console.anthropic.com/account/keys to get one."
+            )
+        self.api_key = api_key
+        self.max_length = max_length
+
+        # Due to reflective construction of all invocation layers we might receive some
+        # unknown kwargs, so we need to take only the relevant.
+        # For more details refer to Anthropic documentation
+        supported_kwargs = ["temperature", "top_p", "top_k", "stop_sequences", "stream", "stream_handler"]
+        self.model_input_kwargs = {k: v for (k, v) in kwargs.items() if k in supported_kwargs}
+
+        # Number of max tokens is based on the official Anthropic model pricing from:
+        # https://cdn2.assets-servd.host/anthropic-website/production/images/FINAL-PRICING.pdf
+        self.max_tokens_limit = 9000
+        self.tokenizer: Tokenizer = self._init_tokenizer()
+
+    def _init_tokenizer(self) -> Tokenizer:
+        # Expire cache after a day
+        expire_after = 60 * 60 * 60 * 24
+        # Cache the JSON config to avoid downloading it each time as it's a big file
+        with requests_cache.enabled(expire_after=expire_after):
+            res = request_with_retry(method="GET", url=CLAUDE_TOKENIZER_REMOTE_FILE)
+            res.raise_for_status()
+        return Tokenizer.from_str(res.text)
+
+    def invoke(self, *args, **kwargs):
+        """
+        Invokes a prompt on the model. It takes in a prompt and returns a list of responses using a REST invocation.
+        :return: The responses are being returned.
+        """
+
+        human_prompt = "\n\nHuman: "
+        assitant_prompt = "\n\nAssistant: "
+
+        prompt = kwargs.get("prompt")
+        if not prompt:
+            raise ValueError(
+                f"No prompt provided. Model {self.model_name_or_path} requires prompt."
+                f"Make sure to provide prompt in kwargs."
+            )
+
+        kwargs_with_defaults = self.model_input_kwargs
+
+        if "stop_sequence" in kwargs:
+            kwargs["stop_words"] = kwargs.pop("stop_sequence")
+        if "max_tokens_to_sample" in kwargs:
+            kwargs["max_length"] = kwargs.pop("max_tokens_to_sample")
+
+        kwargs_with_defaults.update(kwargs)
+
+        # Stream the response either in explicitly specified or if a custom handler is set
+        stream = (
+            kwargs_with_defaults.get("stream", False) or kwargs_with_defaults.get("stream_handler", None) is not None
+        )
+        stop_words = kwargs_with_defaults.get("stop_words", [human_prompt])
```

Comment:
I believe we should ensure that user-specified stop_words should be added to the default Anthropic uses, i.e. we should have a list that contains user-specified stop_words and their default `["\n\nHuman:"]`